### ISSUE TITLE
LND does not run. Same logs repeating with "NAT-PMP" related errors

### ISSUE THREAD

**Issue by samo-sato** [mynodebtc/mynode#883]
**Describe the bug**
Lightning not running on myNode

**Expected behavior**
Lightning running

**Screenshots**
![Screenshot at 2024-11-16 23-27-03](https://github.com/user-attachments/assets/a2db870d-2342-4e11-a7d6-986bd23d0316)

**MyNode hardware (please complete the following information):**
Raspi4B, 8 GB RAM, 1 TB SSD

**Already tried**
- Reboot device
- Upgrade
- Install beta
- Set "nat=false" inside "~/.lnd/lnd.conf"

**Additional context**
- myNode is freshly flashed and installed, because I try to recover myNode from my seed and channel backup, which I already provided to myNode during LN wallet creation using recovery option
- My old myNode stopped working also because Lightning service stopped working - but with different errors (no NAT related as now).
- Bitcoin chain is synced and bitcoin service is running

**LND Logs**
`Nov 16 14:09:32 mynode systemd[1]: lnd.service: Consumed 13.749s CPU time.
Nov 16 14:09:32 mynode systemd[1]: Stopped lnd.service - LND Lightning Daemon.
Nov 16 14:09:32 mynode systemd[1]: lnd.service: Scheduled restart job, restart counter is at 615.
Nov 16 14:09:31 mynode systemd[1]: lnd.service: Consumed 13.749s CPU time.
Nov 16 14:09:31 mynode systemd[1]: lnd.service: Failed with result 'exit-code'.
Nov 16 14:09:31 mynode systemd[1]: lnd.service: Main process exited, code=exited, status=1/FAILURE
Nov 16 14:09:31 mynode lnd[59416]: unable to create server: unable to discover a NAT-PMP enabled device on the local network: Timed out trying to contact gateway
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.521 [INF] LTND: Shutdown complete
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.519 [ERR] LTND: error stopping tor controller: invalid arguments: unexpected code
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.519 [ERR] TORC: DEL_ONION got error: invalid arguments: unexpected code
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.518 [INF] TORC: Stopping tor controller
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.518 [ERR] LTND: Shutting down because error in main method: unable to create server: unable to discover a NAT-PMP enabled device on the local network: Timed out trying to contact gateway
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.518 [ERR] SRVR: unable to discover a NAT-PMP enabled device on the local network: Timed out trying to contact gateway
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.353 [ERR] RPCS: [/lnrpc.Lightning/GetInfo]: the RPC server is in the process of starting up, but not yet ready to accept calls
Nov 16 14:09:31 mynode lnd[59416]: 2024-11-16 14:09:31.342 [ERR] RPCS: [/lnrpc.Lightning/GetInfo]: the RPC server is in the process of starting up, but not yet ready to accept calls
Nov 16 14:09:22 mynode systemd[1]: Started lnd.service - LND Lightning Daemon.
Nov 16 14:09:21 mynode lnd[59416]: 2024-11-16 14:09:21.514 [INF] SRVR: Scanning local network for a NAT-PMP enabled device
Nov 16 14:09:21 mynode lnd[59416]: 2024-11-16 14:09:21.514 [ERR] SRVR: Unable to discover a UPnP enabled device on the local network: multiple NATs detected
Nov 16 14:09:18 mynode lnd[59416]: 2024-11-16 14:09:18.281 [INF] LNWL: Started rescan from block 000000000000000000048403a0c2f35b3f60d38a84dbad9d26d1b19dcad2d62b (height 791938) for 0 addresses
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.282 [INF] SRVR: Scanning local network for a UPnP enabled device
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.282 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.281 [INF] HSWC: Restoring in-memory circuit state from disk
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.281 [INF] HSWC: Finished cleaning: no closed channels found, no actions taken.
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.281 [INF] HSWC: Cleaning circuits from disk for closed channels
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.275 [INF] TORC: Starting tor controller
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.275 [INF] SRVR: Onion services are accessible via Tor! NOTE: Traffic to clearnet services is not routed via Tor.
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.265 [INF] CHRE: LightningWallet opened
Nov 16 14:09:17 mynode lnd[59416]: 2024-11-16 14:09:17.255 [INF] LNWL: The wallet has been unlocked without a time limit
Nov 16 14:09:14 mynode lnd[59416]: 2024-11-16 14:09:14.998 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on 127.0.0.1:28332
Nov 16 14:09:14 mynode lnd[59416]: 2024-11-16 14:09:14.998 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on 127.0.0.1:28333
Nov 16 14:09:14 mynode lnd[59416]: 2024-11-16 14:09:14.998 [INF] CHRE: Initializing bitcoind backed fee estimator in CONSERVATIVE mode
Nov 16 14:09:14 mynode lnd[59416]: 2024-11-16 14:09:14.977 [INF] CHRE: Primary chain is set to: bitcoin
Nov 16 14:09:14 mynode lnd[59416]: 2024-11-16 14:09:14.828 [INF] LNWL: Opened wallet
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.588 [INF] LTND: Attempting automatic wallet unlock with password provided in file
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.588 [INF] LTND: Database(s) now open (time_to_open=9.552342ms)!
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.588 [INF] CHDB: Checking for optional update: prune_revocation_log=false, db_version=empty
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.588 [INF] CHDB: Checking for schema update: latest_version=31, db_version=31
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.580 [INF] LTND: Creating local graph and channel state DB instances
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.580 [INF] CHDB: Not compacting database file at /home/bitcoin/.lnd/data/graph/mainnet/sphinxreplay.db, it was last compacted at 2024-11-16 03:17:21.120412395 -0600 CST (10h51m51s ago), min age is set to 168h0m0s
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.579 [INF] CHDB: Not compacting database file at /home/bitcoin/.lnd/data/chain/bitcoin/mainnet/macaroons.db, it was last compacted at 2024-11-16 03:17:21.074179479 -0600 CST (10h51m51s ago), min age is set to 168h0m0s
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.579 [INF] CHDB: Not compacting database file at /home/bitcoin/.lnd/data/graph/mainnet/channel.db, it was last compacted at 2024-11-16 03:17:21.025608963 -0600 CST (10h51m51s ago), min age is set to 168h0m0s
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.579 [INF] LTND: Opening bbolt database, sync_freelist=false, auto_compact=true
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.578 [INF] LTND: Opening the main database, this might take a few minutes...
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.578 [INF] RPCS: gRPC proxy started at 0.0.0.0:10080
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.562 [INF] RPCS: RPC server listening on 0.0.0.0:10009
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.560 [INF] LTND: Active chain: Bitcoin (network=mainnet)
Nov 16 14:09:12 mynode lnd[59416]: 2024-11-16 14:09:12.560 [INF] LTND: Version: 0.17.2-beta commit=v0.17.2-beta, build=production, logging=default, debuglevel=info
Nov 16 14:09:12 mynode lnd[59416]: Automatically obtained bitcoind's RPC credentials
Nov 16 14:09:12 mynode lnd[59416]: Attempting automatic RPC configuration to bitcoind
Nov 16 14:09:12 mynode lnd[59405]: + exit 0
Nov 16 14:09:12 mynode lnd[59406]: }
Nov 16 14:09:12 mynode lnd[59406]:   "warnings": ""
Nov 16 14:09:12 mynode lnd[59406]:   "pruned": false,
Nov 16 14:09:12 mynode lnd[59406]:   "size_on_disk": 700305270174,
Nov 16 14:09:12 mynode lnd[59406]:   "chainwork": "00000000000000000000000000000000000000009a9992533dbe35eec3981ea2",
Nov 16 14:09:12 mynode lnd[59406]:   "initialblockdownload": false,
Nov 16 14:09:12 mynode lnd[59406]:   "verificationprogress": 0.9999974156940304,
Nov 16 14:09:12 mynode lnd[59406]:   "mediantime": 1731783537,
Nov 16 14:09:12 mynode lnd[59406]:   "time": 1731787229,
Nov 16 14:09:12 mynode lnd[59406]:   "difficulty": 101646843652785.2,
Nov 16 14:09:12 mynode lnd[59406]:   "bestblockhash": "0000000000000000000230d9c0931ea7132d89eec381e6b0599590d58656bbf8",
Nov 16 14:09:12 mynode lnd[59406]:   "headers": 870609,
Nov 16 14:09:12 mynode lnd[59406]:   "blocks": 870609,
Nov 16 14:09:12 mynode lnd[59406]:   "chain": "main",
Nov 16 14:09:12 mynode lnd[59406]: {
Nov 16 14:09:12 mynode lnd[59405]: + bitcoin-cli -datadir=/mnt/hdd/mynode/bitcoin -rpcwait getblockchaininfo
Nov 16 14:09:12 mynode lnd[59405]: + '[' '!' -f /mnt/hdd/mynode/.mynode_bitcoin_synced ']'
Nov 16 14:09:12 mynode lnd[59405]: Checking if Bitcoin is synced...
Nov 16 14:09:12 mynode lnd[59405]: + echo 'Checking if Bitcoin is synced...'
Nov 16 14:09:12 mynode lnd[59405]: + set -e
Nov 16 14:09:12 mynode lnd[59402]: Not shutting down!
Nov 16 14:09:12 mynode lnd[59402]: Checking if device is shutting down...
Nov 16 14:09:12 mynode systemd[1]: Starting lnd.service - LND Lightning Daemon...`

**Comment by tehelsper** [comment]
Swap LND to Tor-only in the networking section of the settings page. Remove your custom LND config.

**Comment by samo-sato** [comment]
> Swap LND to Tor-only in the networking section of the settings page. Remove your custom LND config.

Thanks. I did that and it LND is now running.

I have also found following option (which I did not use) on myNode's Settings page:
**Allow All Local Network Traffic**
`This will allow all local network traffic from 10.0.0.0/8, 192.168.0.0/16, and 172.16.0.0/12. This can be helpful if you are having NAT-PMP or UPnP issues with LND. This will reboot the device.`
Which seems to relate to my original "NAT" issue (based on logs).
